tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/intersection/IntersectionTypeTest.java

```diff
@@ -86,6 +86,8 @@ public void testImmutableTypesNegative() {
         validateError(result, index++, "invalid intersection type '(Baz & readonly)': no intersection", 32,
                       45);
         validateError(result, index++, "incompatible types: 'Y' is not a record", 42, 6);
+        validateError(result, index++, "incompatible types: expected " +
```

Comment:
May make sense to add this test in org.ballerinalang.test.types.readonly.SelectivelyImmutableTypeTest instead.

Reply:
Updated in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/41843/commits/6f8cd3857971872ba1c12e95f60ebb0d4709452b).